### PR TITLE
fix: optimization "createPlaneSegments" algorithm stage

### PR DESCRIPTION
**Pull request by LevDenisov**

### CODE REVIEW

File: cpp/deplex/src/deplex/plane_extractor.cpp [hunk 1]
```diff
@@ -73,6 +74,7 @@ class PlaneExtractor::Impl {
   int32_t image_height_;
   int32_t image_width_;
   Eigen::MatrixXi labels_map_;
+  std::vector<std::vector<Eigen::Index>> neighbours;
```
Comment (achains) [inline cpp/deplex/src/deplex/plane_extractor.cpp]: 1. Why do we need precomputed `neighbours` 
2. Why is it a field of `PlaneExtractor`? It doesn't make sense for me.

File: cpp/deplex/src/deplex/plane_extractor.cpp [hunk 2]
```diff
@@ -303,20 +316,17 @@ std::vector<CellSegment> PlaneExtractor::Impl::createPlaneSegments(CellGrid cons
...
 
     // 4. Merge activated cells & remove from hist
-    for (size_t i = 0; i < activation_map.size(); ++i) {
-      if (activation_map[i]) {
-        plane_candidate += cell_grid[i];
-        hist.removePoint(static_cast<int32_t>(i));
-        unassigned_mask[i] = false;
-        --remaining_planar_cells;
-      }
+    for (auto& v : cells_to_merge) {
```
Comment (achains) [inline cpp/deplex/src/deplex/plane_extractor.cpp]: Using reference for primitive type (e.g. int) is more expensive than using its copy. It's better to remove `&`

File: cpp/deplex/src/deplex/plane_extractor.cpp [hunk 3]
```diff
@@ -326,34 +336,31 @@ std::vector<CellSegment> PlaneExtractor::Impl::createPlaneSegments(CellGrid cons
...
-      // TODO: Effective assigning by mask?
-      int stacked_cell_id = 0;
-      for (int32_t row_id = 0; row_id < nr_vertical_cells_; ++row_id) {
-        for (int32_t col_id = 0; col_id < nr_horizontal_cells_; ++col_id) {
-          if (activation_map[stacked_cell_id]) {
-            labels_map_.row(row_id)[col_id] = nr_curr_planes;
-          }
-          ++stacked_cell_id;
-        }
+      for (auto& v : cells_to_merge) {
```
Comment (achains) [inline cpp/deplex/src/deplex/plane_extractor.cpp]: Same. Remove `&`

**Review by achains — CHANGES_REQUESTED** [review]

File: cpp/deplex/src/deplex/plane_extractor.cpp (hunk 1 above)
Comment (LevDenisov) [reply cpp/deplex/src/deplex/plane_extractor.cpp]: To get rid of numerous calls to the getNeighbours function, in which we look at neighboring cells every time and initialize a vector with them, when processing each image

**Review by achains — APPROVED** [review]

File: cpp/deplex/src/deplex/cell_grid.cpp [hunk 4]
```diff
@@ -15,6 +15,7 @@
  */
 #include "cell_grid.h"
 
+#include <iostream>
```
Comment (achains) [inline cpp/deplex/src/deplex/cell_grid.cpp]: Remove iostream

**Review by achains — CHANGES_REQUESTED** [review]

**Review by achains — APPROVED** [review]

**Pull request merged**